### DISCUSSION TITLE
How to slow down turnstile check?

### DISCUSSION THREAD

**Discussion by sfxter** [ZFC-Digital/puppeteer-real-browser#214] [Q&A]
I have noticed, that when Captcha checkbox is clicked too fast, it does not provide clearance cookie. Any ways to add delay before solving captcha? Thank you!

**Comment by RubberArchind** [comment]
i dont think that's the problem,  can you share the website url where you found this behaviour?

because i never experience this even when it click the captcha checkbox after the page just loaded it always return the cookie

**Reply by sfxter** [reply]
I added 1 second delay to the turnstile.js code before click command and it helped me get cf_clearance cookie most of the time. How ever, it still does not guarantees the clearance cookie. I have found, that setting up suspicious user agent helps to trigger captcha and get clearance cookie.

**Reply by sfxter** [reply]
looks like you are right - clearance not depends on the turnstile delay (at least today). Looks like CF have some kind of schedule, where they change when to send you clearance cookie and when not to. And of course, it also depends on the IP that is being used for request. If request comes from outside EU (for example Morroco), it gives clearance cookie. But if request comes from Germany or Poland, most of the time cookie is not given. I am very confused.

**Reply by RubberArchind** [reply]
interesting, what proxy that you used? is that shared proxy or dedicated?

**Comment by RubberArchind** [comment]
anyway have you try this? https://github.com/zfcsoftware/cf-clearance-scraper

**Reply by sfxter** [reply]
will have a look, thank you!
